src/cmclient/cmclient.hpp

```diff
@@ -107,12 +111,16 @@ class CMClient : public aos::sm::launcher::InstanceStatusReceiverItf,
     aos::sm::launcher::LauncherItf*                              mLauncher = {};
     ResourceManager*                                             mResourceManager = {};
     DownloadReceiverItf*                                         mDownloader = {};
-    aos::Thread<>                                                mThread = {};
+    aos::Thread<>                                                mThreadWriter = {};
+    aos::Thread<>                                                mThreadReader = {};
     atomic_t                                                     mState = {};
-    vch_handle                                                   mSMVChanHandler;
+    bool                                                         mWriteFail = {};
+    vch_handle                                                   mSMVChanHandlerWriter;
+    vch_handle                                                   mSMVChanHandlerReader;
     servicemanager_v3_SMIncomingMessages                         mIncomingMessage;
     servicemanager_v3_SMOutgoingMessages                         mOutgoingMessage;
     aos::Mutex                                                   mMutex;
+    aos::ConditionalVariable                                     mWaitWriteFail {mMutex};
```

Comment:
It is used not only for write fail indication but for wake up write thread. Consider renaming it accordingly: `mWriterCondVar` for example.

Reply:
Done